jbpm-console-ng-process-runtime/jbpm-console-ng-process-runtime-client/src/test/java/org/jbpm/console/ng/pr/client/test/list/ProcessInstanceListPresenterTest.java

```diff
@@ -0,0 +1,113 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.anyObject;
+import org.mockito.Mock;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.when;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+@RunWith(GwtMockitoTestRunner.class)
+public class ProcessInstanceListPresenterTest {
+    
+    @Mock
+    private DataSetQueryHelper dataSetQueryHelperMock;
+    
+    @Mock
+    private DataSetQueryHelper dataSetQueryHelperDomainSpecificMock;
+    
+    @Mock
+    private FilterSettings filterSettingsMock;
+    
+    @Mock
+    private DataSetProcessInstanceWithVariablesListView viewMock;
+    
+    @Mock
+    private ExtendedPagedTable<ProcessInstanceSummary> tableMock;
+    
+    @Mock
+    private DataSet dataSetMock;
+    
+
+
+    private MyDataSetProcessInstanceWithVariablesListPresenter presenter;
+
+    
+    @Before
+    public void setupMocks() {
+
+        when(dataSetQueryHelperMock.getCurrentTableSettings()).thenReturn(filterSettingsMock);
+        when(viewMock.getListGrid()).thenReturn(tableMock);
+        when(tableMock.getPageSize()).thenReturn(10);
+        when(dataSetMock.getRowCountNonTrimmed()).thenReturn(8);
+        when(viewMock.getVariablesTableSettings("mock")).thenReturn(filterSettingsMock);
+        
+        //Mock that actually calls the callbacks
+        doAnswer(new Answer() {
+
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+                ((DataSetReadyCallback)invocation.getArguments()[1]).callback(dataSetMock);
+                return null;
+            }
+        }).when(dataSetQueryHelperMock).lookupDataSet(anyInt(), (DataSetReadyCallback) anyObject());
+        
+        //Mock that actually calls the callbacks
+        doAnswer(new Answer() {
+
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+                ((DataSetReadyCallback)invocation.getArguments()[1]).callback(dataSetMock);
+                return null;
+            }
+        }).when(dataSetQueryHelperDomainSpecificMock).lookupDataSet(anyInt(), (DataSetReadyCallback) anyObject());
+        
+    }
+
+    @Test
+    public void domainSpecificColumnsWithPagination() {
+        presenter = new MyDataSetProcessInstanceWithVariablesListPresenter(viewMock, dataSetQueryHelperMock, dataSetQueryHelperDomainSpecificMock); 
+        presenter.setDataSetQueryHelperMock(dataSetQueryHelperMock);
+        presenter.setFilterSettingsMock(filterSettingsMock);
+        presenter.setDataSetQueryHelperDomainSpecificMock(dataSetQueryHelperDomainSpecificMock);
+        //Get the first page of process instances selected
+        Range range = new Range(0, 10);
+        
+        presenter.getData(range);
+     
+    }
+    
```

Comment:
@Salaboy ah I see, never mind, you moved this to the MyDataSetProcessInstanceWithVariablesListPresenter.